stumpy/stump.py

```diff
@@ -201,14 +201,13 @@ def _compute_diagonal(
 
             if T_B_subseq_isfinite[uint64_j] and T_A_subseq_isfinite[uint64_i]:
                 # Neither subsequence contains NaNs
-                if T_B_subseq_isconstant[uint64_j] or T_A_subseq_isconstant[uint64_i]:
+                if T_B_subseq_isconstant[uint64_j] and T_A_subseq_isconstant[uint64_i]:
+                    pearson = 1.0
+                elif T_B_subseq_isconstant[uint64_j] or T_A_subseq_isconstant[uint64_i]:
                     pearson = 0.5
                 else:
                     pearson = cov * Σ_T_inverse[uint64_j] * σ_Q_inverse[uint64_i]
```

Comment:
we may add:
```
if pearson > 1.0:
   pearson = 1.0
```
here. And, then we can later remove `abs(...)` when we want to convert pearson to distance in `_stump`

Reply:
I like this suggestion. However, if there is little cost in keeping `abs`, I would like to keep it as security? Maybe I'm being overly paranoid 😆

Reply:
Sure.  Even if we change something here, we have `abs(...)`  that can protect us later. So, I think what you are saying is reasonable. Thank you for the suggestion :)